src/OrchardCore/OrchardCore.Abstractions/Localization/CultureScope.cs

```diff
@@ -1,45 +1,55 @@
 using System;
 using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Environment.Shell.Scope;
 
-namespace OrchardCore.Localization
+namespace OrchardCore.Localization;
+
+public sealed class CultureScope : IDisposable
 {
-    public sealed class CultureScope : IDisposable
+    private readonly CultureInfo _originalCulture;
+    private readonly CultureInfo _originalUICulture;
+
+    private CultureScope(CultureInfo culture, CultureInfo uiCulture)
     {
-        private readonly CultureInfo _originalCulture;
-        private readonly CultureInfo _originalUICulture;
+        Culture = culture;
+        UICulture = uiCulture;
+        _originalCulture = CultureInfo.CurrentCulture;
+        _originalUICulture = CultureInfo.CurrentUICulture;
 
-        private CultureScope(CultureInfo culture, CultureInfo uiCulture)
-        {
-            Culture = culture;
-            UICulture = uiCulture;
-            _originalCulture = CultureInfo.CurrentCulture;
-            _originalUICulture = CultureInfo.CurrentUICulture;
+        SetCultures(culture, uiCulture);
+    }
 
-            SetCultures(culture, uiCulture);
-        }
+    public CultureInfo Culture { get; }
 
-        public CultureInfo Culture { get; }
+    public CultureInfo UICulture { get; }
 
-        public CultureInfo UICulture { get; }
+    public static CultureScope Create(string culture) => Create(culture, culture);
 
-        public static CultureScope Create(string culture) => Create(culture, culture);
+    public static CultureScope Create(string culture, string uiCulture) => CreateInternal(culture, uiCulture);
 
-        public static CultureScope Create(string culture, string uiCulture)
-            => Create(CultureInfo.GetCultureInfo(culture), CultureInfo.GetCultureInfo(uiCulture));
+    public static CultureScope Create(CultureInfo culture) => Create(culture, culture);
 
-        public static CultureScope Create(CultureInfo culture) => Create(culture, culture);
+    public static CultureScope Create(CultureInfo culture, CultureInfo uiCulture) => CreateInternal(culture.Name, uiCulture.Name);
 
-        public static CultureScope Create(CultureInfo culture, CultureInfo uiCulture) => new CultureScope(culture, uiCulture);
+    public void Dispose()
+    {
+        SetCultures(_originalCulture, _originalUICulture);
+    }
 
-        public void Dispose()
-        {
-            SetCultures(_originalCulture, _originalUICulture);
-        }
+    private static void SetCultures(CultureInfo culture, CultureInfo uiCulture)
+    {
+        CultureInfo.CurrentCulture = culture;
+        CultureInfo.CurrentUICulture = uiCulture;
+    }
+
+    private static CultureScope CreateInternal(string culture, string uiCulture)
+    {
+        var useUserSelectedCultureSettings = ShellScope.Services.GetService<ILocalizationService>()
```

Comment:
@jtkech is there a better way than using `ShellScope.Services`?

Reply:
That's okay for me to resolve a service like this but only when we are sure it is always used in the context of a valid ShellScope.

Then here you have to use .GetAwaiter().GetResult() which is not good, you would need to add an async `CultureScope.CreateAsync()`.

That said here I would have thought about an additional parameter `useUserOverride` that would be passed by the consumer. Also not sure the option should come from admin settings, maybe better to use a configurable IOptions.

Reply:
> but only when we are sure it is always used in the context of a valid ShellScope

How can I verify?

> That said here I would have thought about an additional parameter useUserOverride that would be passed by the consumer. Also not sure the option should come from admin settings, maybe better to use a configurable IOptions.

If you look to the commits history I already used `IOptions<T>` then I move into `ILocalizationService`

Did you think I need to follow the same pattern that I did in `CultureScopeTests` in other broken tests?

@Piedone any though coz `CultureScope` doesn't accept constructor injection

Reply:
> but only when we are sure it is always used in the context of a valid ShellScope

How can I verify?

> That said here I would have thought about an additional parameter useUserOverride that would be passed by the consumer. Also not sure the option should come from admin settings, maybe better to use a configurable IOptions.

If you look to the commits history I already used `IOptions<T>` then I move into `ILocalizationService`

Did you think I need to follow the same pattern that I did in `CultureScopeTests` in other broken tests?

@Piedone any though coz `CultureScope` doesn't accept constructor injection

Reply:
> That said here I would have thought about an additional parameter useUserOverride that would be passed by the consumer

This is could be a simple option, but I was thinking about centeral place that fetch the culture settings either from `appsettings.json` or site settings

Reply:
Yep, I'd use `IOptions` too.

Reply:
If you want to go with `IOptions` as I did at the beginning, I will rollback from the latest changes